### DISCUSSION TITLE
Welcome to STDFReader Discussions!

### DISCUSSION THREAD

**Discussion by showjim** [showjim/STDFReader#2] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by ericstanleysamar** [comment]
Thank you for this! As someone who works in a semicon industry, I find this very useful.
Is it okay if I added customized functions for this for myself?

**Reply by showjim** [reply]
Thank you for you like my tool! You can fork my repo and then it is OK for you to modify it.

**Comment by noonchen** [comment]
Hi Chao, I'm the author of STDF-Viewer and also worked in a semicon company. Thanks to your recent commit about the MPR handling, it really helped me understand this type of record. Due to the limitations of stdf file sources, I don't have the chance to add the support for MPR in STDF-Viewer, could you share some stdf files contain MPR so that I can try to visualize this record in my tool? ;)

**Reply by showjim** [reply]
Hi Noon,
I am glad that my project can help you,  STDF-Viewer is an excellent tool with much more beautiful UI than mine~
The STDFs with MPR I used are from customer side so I cannot share them to you, but I have generated one demo STDF with MPR offline, and have send it to you by email.

**Reply by noonchen** [reply]
Wow, much appreciated!

**Comment by franli0** [comment]
Hi Jerry, I'm trying to analyze stdf files recently and I found your awesome work. However, I ran into an issue when I tried to upload stdf files. I started the app from the command line `python STDF_Reader_GUI.py` and it went well at the first attempt, but when I tried to choose another stdf file it returns `QObject::~QObject: Timers cannot be stopped from another thread`, and the app crashed. This issue didn't appear when I chose the sample file you provide. Do you have any suggestions on how to use the app? I ran the app on Windows 7 32bit.
![image](https://user-images.githubusercontent.com/19802431/131635529-9e864aae-937f-4540-a9b6-dfec5f745e9e.png)

**Reply by showjim** [reply]
Hi Franlio, it seems a error or warning from PyQt5 which I did not encounter before. Could you share the stdf file that can reproduce this message to me for debugging?

**Reply by franli0** [reply]
Hi Jerry, sorry for the late update. Apparently, the issue was caused by the stdf file. After I eliminated a certain section in the stdf file, the app worked without any issues.

**Comment by JulianZhao1025** [comment]
<img width="1128" height="111" alt="image" src="https://github.com/user-attachments/assets/f3f4daa9-f056-4a95-8885-e6c394e93adf" />
Hi jerry，

    I installed this lib on a Linux system that was not connected to the internet. However, when I used it, I encountered this error. Could you please help me figure out what caused this?
    thanks.

**Reply by showjim** [reply]
Hi @JulianZhao1025 , could you share a demo stdf which can lead to this error?

**Reply by JulianZhao1025** [reply]
<img width="632" height="129" alt="image" src="https://github.com/user-attachments/assets/c4cfcc10-7ff6-4227-a842-9f251d988c8f" />
Hii Jerry，

I tried many STDFs, and the results were all the same. 
It seems that this is not a problem caused by a specific STDF, but rather an issue with the software environment. 
So I'm more curious about what kind of environment is missing.

**Reply by JulianZhao1025** [reply]
If the analysis is too difficult, don't worry, it won't take up too much of your time. If that's the case, just analyze it on Windows.

**Reply by showjim** [reply]
I have tested **stdf-reader** in clean environments on both **Windows 11** and **Ubuntu 24.04** with **Python 3.12.3** and was unable to reproduce the issue you reported. Since I also use this tool in my day-to-day work without encountering similar problems, I don't think we can definitively attribute the issue to the environment at this stage. We should first rule out potential problems with the STDF file itself.

Without a sample STDF file, it's difficult for me to investigate further or determine the root cause of the behavior you're seeing. If you're unable to share a sample, my ability to analyze the issue will be very limited.
<img width="728" height="144" alt="image" src="https://github.com/user-attachments/assets/1318b2df-0839-4ded-8e4e-1ed212356530" />

**Reply by JulianZhao1025** [reply]
<img width="1001" height="144" alt="image" src="https://github.com/user-attachments/assets/6f42ff60-a9fb-4adf-9bee-4288cbcf5306" />

The stdf I'm using is the example file you provided, namely a595.stdf. All these sdf files are unable to be converted. My Linux system is: RED HAT 7.9

**Reply by showjim** [reply]
<img width="614" height="147" alt="image" src="https://github.com/user-attachments/assets/6d39156a-bb7c-4027-9793-3322c727fd89" />

The attached screenshot shows that a595.stdf can be parsed correctly in my Ubuntu environment. Since I don't have access to a Red Hat system, I'm unable to verify the behavior there. It may be worth testing on another Linux distribution to see whether the issue is specific to the Red Hat environment.

**Reply by JulianZhao1025** [reply]
Hi Jerry，

The problem has been solved! The main reason is that under Linux, pystdf does not support passing parameters for "skiptype". I modified the code of stdf_reader (as shown in the figure) and it can now be used. Thank you for your support.

<img width="444" height="82" alt="image" src="https://github.com/user-attachments/assets/18af4c00-718f-4562-81bf-ecb388495bf4" />
<img width="668" height="138" alt="image" src="https://github.com/user-attachments/assets/7764c8cc-61a0-4a22-a6a5-9ea76e096250" />

**Reply by showjim** [reply]
This reminds that I use a modified pystdf version (to support STDF V4-2007.1), if you want to use the full features of stdf-reader, you need to uninstall the original pystdf and re-install stdf-reader.

**Reply by JulianZhao1025** [reply]
Following your suggestion, I reinstalled it and it worked successfully. Thank you very much for your reply!
